Objective-C/CBLCollection.mm

```diff
@@ -169,26 +170,101 @@ - (BOOL) deleteDocument: (CBLDocument*)document
     return NO;
 }
 
-- (CBLDocument*) documentWithID: (NSString*)docID error: (NSError**)error {
-    // TODO: add implementation
-    return nil;
+- (CBLDocument*) documentWithID: (NSString*)documentID error: (NSError**)error {
```

Comment:
assert documentID not nil

Reply:
I guess, we make the change in separate PR. If we move all calls from Database to use the Collection API, which might make this a huge PR. I will add a TODO to do same.